src/platforms/stm32/src/CMakeLists.txt

```diff
@@ -30,6 +30,28 @@ if(CMAKE_COMPILER_IS_GNUCC)
 endif()
 
 add_subdirectory(../../../libAtomVM libAtomVM)
+if(CMAKE_COMPILER_IS_GNUCC)
+    target_compile_options(libAtomVM PUBLIC
+    -Wno-bad-function-cast
```

Comment:
Do we really want to suppress the warnings? For example bad function casts can crash on emscripten platform.

Reply:
I am sure I was a little heavy handed with the warnings that were suppressed. I do not understand why none of the other ports emit any where near the number of warnings from libAtomVM.

The actual number of compiler warnings is over 80,000, it isn't just difficult, it is impossible to spot new errors that might be introduced to the stm32 port code itself.

Reply:
The bad function cast is something that should be resolved... I would appreciate help  selecting any important warnings that should not be suppressed, I can then open issues for those specific problems.  Until the number of warning is reduced findings those serious ones is nearly impossible.

Reply:
All warnings are useful but pedantic, that might be too much.
The main issue with pedantic is that it complains about few GNU C extensions that are quite widespread.

Reply:
Unfortunately pedantic only accounts for an extremely small number of the warnings. Suppressing those alone would not make any noticeable reduction in warnings.